### ISSUE TITLE
GlobaLeaks 2.70 handling of Tor HS loading with an incompatible version of Tor

### ISSUE THREAD

**Issue by fpietrosanti** [globaleaks/GlobaLeaks#1989]
**Current behavior**

When GlobaLeaks with the new 2.70 edition, try to load a TorHS descriptor via the Tor Control Port trough it's unix socket, if the Tor version is not at least =>2.9 it will get an unhandled exception:

```
GlobaLeaks version: 2.70.1

txtorcon.torcontrolprotocol.TorProtocolError 
    Happens on 500-level responses in the protocol, almost certainly
    in an errback chain.

    :ivar code: the actual error code
    :ivar text: other text from the protocol
    

TorProtocolError: 510 Unrecognized command "ADD_ONION"
```

**Expected behavior**
This ticket is to:
a. Handle this unhandled exception from txtorcon by properly logging it and reporting it in the user interface
b. Handle the forceful setup of Tor 2.9 as a minimum requirement of GlobaLeaks, either in new installation either during upgrades
c. Ensure that the Tor's debian repository is properly and forcefully setup

**Comment by fpietrosanti** [comment]
@evilaliv3 so that's fixed with new release, that prevent Tor problem and detect in a handled way the error if it happens?

**Comment by evilaliv3** [comment]
@fpietrosanti:

- in ticket https://github.com/globaleaks/GlobaLeaks/issues/1994 we addressed the topic of handling with retries and failures logging.
- the install script has been fixed in order to perform Tor repository setup and to allow multiple executions in order to fix old setups

in my opinion this is the more that we can do; any other approach dealing with looking the versions at runtime on the system, fixing repositories from within a debian package or other quirks are out of any good practice for packaging so would not suggest to go for it.

this is why i would consider to close this ticket with what done.

\cc @NSkelsey
